src/alpaka/AlpakaCore/alpakaConfig.h

```diff
@@ -1,154 +1,7 @@
 #ifndef alpakaConfig_h_
 #define alpakaConfig_h_
 
-#include <alpaka/alpaka.hpp>
```

Comment:
Why was it useful to split this file?

Reply:
There is now a source file in `src/alpaka/AlpakaCore`, which implied having a `src/alpaka/AlpakaCore/alpaka` repo to be able to compile this with -DALPAKA_ACC_GPU_CUDA_ENABLED etc:
```
 #ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
#define ALPAKA_ACC_GPU_CUDA_ASYNC_BACKEND
namespace alpaka_cuda_async {
  using namespace alpaka_common;
  using Acc1 = alpaka::AccGpuCudaRt<Dim1, Extent>;
  using Acc2 = alpaka::AccGpuCudaRt<Dim2, Extent>;
  using Queue = alpaka::QueueCudaRtNonBlocking;
}  // namespace alpaka_cuda_async
```

Reply:
Ok, I don't follow though how `src/alpaka/AlpakaCore/alpaka/alpakaDevAcc.cc` would not work with the monolithic `alpakaConfig.h`.

Reply:
For each package:
- Source placed in alpaka subdirectory (`alpaka/AlpakaCore/alpaka`) is compiled with -DALPAKA_ACC_GPU_CUDA_ENABLED, etc, for each accelerator type.
- Source directly placed in the package (`alpaka/AlpakaCore/`) is not compiled for each backend (source common for all backends).

Here, `alpakaDevAcc.cc` needs to be placed in the alpaka subdirectly (because it is compiled for each backend).
Instead, alpakaDevHost.cc cannot be placed in the alpaka subdirectory: would not make sense, as it is common for all backends, and actually the linker complains about it. 
So I put `alpakaDevHost.cc` in `src/alpaka/AlpakaCore`.
In my opinion, this makes sense: all common code is in `src/alpaka/AlpakaCore`, and all accelerator-dependent code is in `src/alpaka/AlpakaCore/alpaka`.

Both `alpakaDevHost.cc` and `alpakaDevAcc.cc` need to include `src/alpaka/AlpakaCore/alpakaConfig.h`.
If `src/alpaka/AlpakaCore/alpakaConfig.h` stays as it is (one file), the compilation of `alpakaDevHost.cc` does not work, because there is no -DALPAKA_ACC_GPU_CUDA_ENABLED etc during compilation (because `alpakaDevHost.cc` is not placed in alpaka subdirectory). 

Hence the split of `src/alpaka/AlpakaCore/alpakaConfig.h` into common utilities  (`alpakaConfigCommon`) and accelerator-dependent utilities ( `alpakaConfigAcc`): `alpakaDevHost` only needs, and only includes, `alpakaConfigCommon`.

Before adding host/device, there was no such issue, because there was not source file in entire `src/alpaka/AlpakaCore`: the headers (including the acc-dependents parts) were directly included from `src/alpaka/my-plugin/alpaka` (where all the alpaka-dependent code is placed).

Reply:
Thanks for the explanation